File: pages/hacker-news.page.ts
import { Page } from '@playwright/test';
import { BasePage } from './base.page';
import { SITES } from '../fixtures/test-data';

export class HackerNewsPage extends BasePage {
  async open() {
    await this.goto(SITES.hackerNews.url);
    await this.waitForStableLayout(1500);
  }

  get mainContent() {
    return this.page.locator('table.itemlist, #hnmain');
  }

  get header() {
    return this.page.locator('tr:first-child');
  }
}

File: pages/hacker_news_page.py
from fixtures.test_data import SITES
from pages.base_page import BasePage


class HackerNewsPage(BasePage):
    def open(self) -> None:
        self.goto(SITES["hacker_news"]["url"])
        self.wait_for_stable_layout(1500)

    @property
    def main_content(self):
        return self.page.locator("table.itemlist, #hnmain")

    @property
    def header(self):
        return self.page.locator("#hnmain .pagetop, #hnmain table:first-of-type tbody tr").first
